src/plonkish/ir/assignments.rs

```diff
@@ -125,18 +137,39 @@ impl<F: Field + Hash, TraceArgs> AssignmentGenerator<F, TraceArgs> {
         }
     }
 
+    /// Create an AssignmentGenerator without a trace generator
+    /// for the case when the trace is provided externally
+    pub fn without_trace_generator(
```

Comment:
I don't think this method should be necessary, how would the AssignmentGenerator work without a TraceGenerator to provide the the TraceWitness?